appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1662,9 +1662,15 @@ private void deployAll() throws GlassFishException {
                         // Convert the URL to a URI for use with the deploy method
                         URI artefactURI = deploymentMapEntry.getValue().toURI();
 
-                        deployer.deploy(artefactURI, "--availabilityenabled",
-                                "true", "--contextroot",
-                                deploymentMapEntry.getKey(), "--force=true", "--loadOnly", "true");
+                        String artefactName= artefactURI.getPath().substring(artefactURI.getPath().lastIndexOf('/') + 1);
+                        // artefact name always has a valid extension
+                        String name = artefactName.substring(0, artefactName.length() - 4);
```

Comment:
1. Could this be extracted to (tested) method?
2. Could this variable be renamed to something like `nameWithExtension` - and above comment removed?